src/Security/EmailVerifier.php

```diff
@@ -34,10 +30,15 @@ public function sendEmailConfirmation(string $verifyEmailRouteName, UserInterfac
             throw new \UnexpectedValueException('Expected '.User::class.', got '.get_class($user));
         }
 
+        $mail = $user->getEmail();
+        if (null === $mail) {
+            throw new \InvalidArgumentException('User needs to have an email address set.');
+        }
```

Comment:
If you rebase those checks aren't needed anymore as getEmail always returns string